precompiles/avs/IAVSManager.sol

```diff
@@ -4,15 +4,43 @@
 address constant AVSMANAGER_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000000901;
 
 /// @dev The avs-manager contract's instance.
-IAVSManager constant AVSMANAGER_CONTRACT = IAVSManager(
-    AVSMANAGER_PRECOMPILE_ADDRESS
-);
-
+IAVSManager constant AVSMANAGER_CONTRACT = IAVSManager(AVSMANAGER_PRECOMPILE_ADDRESS);
 /// @author Exocore Team
 /// @title AVS-Manager Precompile Contract
 /// @dev The interface through which solidity contracts will interact with AVS-Manager
 /// @custom:address 0x0000000000000000000000000000000000000901
+
 interface IAVSManager {
+
+    event AVSRegistered(address indexed avsAddr, string sender, string avsName);
+    event AVSUpdated(address indexed avsAddr, string sender, string avsName);
+    event AVSDeregistered(address indexed avsAddr, string sender, string avsName);
+    event OperatorJoined(address indexed avsAddr, string sender);
+    event OperatorLeft(address indexed avsAddr, string sender);
+    event TaskCreated(
+        string sender,
+        uint64 taskId,
+        address indexed taskContractAddress,
+        string name,
+        bytes hash,
+        uint64 taskResponsePeriod,
+        uint64 taskChallengePeriod,
+        uint64 thresholdPercentage,
+        uint64 taskStatisticalPeriod
+    );
+    event ChallengeInitiated(
+        string sender, bytes taskHash, uint64 taskID, bytes taskResponseHash, string operatorAddress
+    );
+    event PublicKeyRegistered(string sender, string name);
+    event TaskSubmittedByOperator(
+        string sender,
+        uint64 taskID,
+        bytes taskResponse,
+        bytes blsSignature,
+        address indexed taskContractAddress,
+        uint8 phase
+    );
```

Comment:
@coderabbitai: Please refrain from such suggestions. We do not want to add the `indexed` keyword intentionally since it results in the information being hashed irretrievably.